src/client/Client.cc

```diff
@@ -9644,28 +9644,34 @@ Inode *Client::ll_get_inode(vinodeno_t vino)
   return in;
 }
 
-int Client::ll_getattr(Inode *in, struct stat *attr, int uid, int gid)
+int Client::_ll_getattr(Inode *in, int uid, int gid)
 {
-  Mutex::Locker lock(client_lock);
-
   vinodeno_t vino = _get_vino(in);
 
   ldout(cct, 3) << "ll_getattr " << vino << dendl;
   tout(cct) << "ll_getattr" << std::endl;
   tout(cct) << vino.ino.val << std::endl;
 
+  if (vino.snapid < CEPH_NOSNAP)
+    return 0;
+  else
+    return _getattr(in, CEPH_STAT_CAP_INODE_ALL, uid, gid);
+}
+
+int Client::ll_getattr(Inode *in, struct stat *attr, int uid, int gid)
+{
+  Mutex::Locker lock(client_lock);
+
+  int res = _ll_getattr(in, uid, gid);
+  vinodeno_t vino = _get_vino(in);
+
   /* special case for dotdot (..) */
   if (vino.ino.val == CEPH_INO_DOTDOT) {
     attr->st_mode = S_IFDIR | 0755;
     attr->st_nlink = 2;
     return 0;
   }
 
```

Comment:
Are we okay doing this check after _getattr()? Since it can trigger an MDS request?